exe/genv-devices

```diff
@@ -136,32 +164,44 @@ def find_available_device(devices: Devices, ignore: List[int], gpu_memory: Optio
 
 def is_device_available(device: Device, gpu_memory: Optional[str]) -> bool:
     """
-    Checks if a device is available.
-
-    Checking if the device is empty if amount of GPU memory not specified.
-    Otherwise, checking if this amount is available.
+    Check for device is availability.
+    If a memory consumption constraint is specified, check if this amount is available on the device.
+    Otherwise, a device must be non-attached to be considered "available".
+    :param device:
+    :param gpu_memory: memory constraint (optional)
+    :return: True iff the device is available
     """
-    return len(device["eids"]) == 0 if gpu_memory is None else get_device_unallocated_memory_bytes(device) >= genv.memory_to_bytes(gpu_memory)
+    return len(device["eids"]) == 0 if gpu_memory is None else get_device_unallocated_memory_bytes(
+        device) >= genv.memory_to_bytes(gpu_memory)
 
 
 def get_device_unallocated_memory_bytes(device) -> int:
     """
     Get the device unallocated memory in bytes.
+    If a device is attached without specifying memory consumption constraint,
+     then all of the devices memory is considered "allocated".
+    :param device:
+    :return: unallocated memory in bytes.
     """
     unallocated_memory_bytes = genv.memory_to_bytes(device["total_memory"])
 
     for env in device["eids"].values():
         if "gpu_memory" not in env:
-            return 0 # the environment uses all the device memory
+            return 0  # the environment uses all the device memory
         else:
             unallocated_memory_bytes -= genv.memory_to_bytes(env["gpu_memory"])
 
-    return max(unallocated_memory_bytes, 0) # convert negative values when device is overallocated
+    return max(unallocated_memory_bytes, 0)  # convert negative values when device is overallocated
 
 
 def find_available_devices(devices: Devices, count: int, ignore: List[int], gpu_memory: Optional[str]) -> List[int]:
     """
-    Finds a number of available devices.
+    Find count (or less, if not enough available devices exists) available devices.
```

Comment:
I think this function will not find _less_ than `count` but fail with an exception
wdyt?

Reply:
I think this is a good Idea, but making it in a seperate PR would be better

Reply:
I actually mean that the documentation is not correct because the function does not find less than `count` as documented, but it will raise `RuntimeError` just like `find_available_device` (which is properly documented 💪)

Reply:
You are corrent :)
I fixed the documentation here to:
Find count available devices, considering blacklist and memory constraints.
     If not enough devices are available, RuntimeError error will be raised.